addons/account/migrations/12.0.1.1/post-migration.py

```diff
@@ -235,6 +235,37 @@ def populate_fiscal_years(env):
                 create_date, create_uid, write_date, write_uid
             FROM account_fiscalyear;
             """)
+    elif openupgrade.table_exists(env.cr, 'date_range_type') and \
+            openupgrade.column_exists(env.cr, 'date_range_type', 'fiscal_year'):
+        openupgrade.logged_query(
+            env.cr, """
+            INSERT INTO account_fiscal_year (
+                name, company_id, date_from, date_to,
+                create_date, create_uid, write_date, write_uid
+            )
+            SELECT
+                dr.name, COALESCE(dr.company_id, drt.company_id),
+                dr.date_start, dr.date_end,
+                dr.create_date, dr.create_uid, dr.write_date, dr.write_uid
+            FROM date_range dr
+            JOIN date_range_type drt ON dr.type_id = drt.id
+            WHERE drt.fiscal_year AND COALESCE(dr.company_id, drt.company_id) IS NOT NULL
+            """)
+        for company in env["res.company"].search([]):
```

Comment:
Don't do loop but do JOIN with res_company in the SQL query, and please reuse the common query part putting it in a string variable for more readable code.

Reply:
How to do the join of res_company?

Reply:
With a CROSS JOIN

Reply:
Done :)